packages/@uppy/aws-s3-multipart/src/index.js

```diff
@@ -348,14 +348,36 @@ class HTTPCommunicationQueue {
   async uploadChunk (file, partNumber, chunk, signal) {
     throwIfAborted(signal)
     const { uploadId, key } = await this.getUploadId(file, signal)
-    throwIfAborted(signal)
+
+    const signatureRetryIterator = this.#retryDelays.values()
+    const chunkRetryIterator = this.#retryDelays.values()
```

Comment:
This change makes it so the retry iterator will be reset for each chunk, i.e. if there's a failure on one chunk, it won't affect the other chunks. I don't think it's a good change.

Reply:
I thought this is what we wanted 🤔 

Let's say we upload 5K chunks, then only 5 chunks in total are allowed to fail? Why shouldn't any chunk be allowed to fail and retry?

Reply:
IMO it makes sense to increase the time before retry if any chunk has failed, otherwise we increase the chances of running into rate limiting issues. We could also just go with it and see if anyone complains.